girder/molecules/molecules/utilities/molecules.py

```diff
@@ -30,22 +30,13 @@ def create_molecule(data_str, input_format, user, public, gen3d=True,
                     provenance='uploaded by user'):
 
     using_2d_format = (input_format in openbabel_2d_formats)
-    smiles_format = 'smiles'
+    inchi_format = 'inchi'
 
     if using_2d_format or input_format in openbabel_3d_formats:
-        smiles = openbabel.to_smiles(data_str, input_format)
+        inchi, inchikey = openbabel.to_inchi(data_str, input_format)
     else:
         sdf_data = avogadro.convert_str(data_str, input_format, 'sdf')
-        smiles = openbabel.to_smiles(sdf_data, 'sdf')
-
-    props = openbabel.properties(smiles, smiles_format)
-    atom_count = props['atomCount']
-
-    if atom_count > 1024:
-        raise RestException('Unable to generate inchi, '
-                            'molecule has more than 1024 atoms .', code=400)
```

Comment:
Because we are converting to InChI right away now, it might be tricky to find the number of atoms before generating the InChI. I thought that maybe we could just let Open Babel fail and raise an exception if there are more than 1024 atoms.

However, I gave it a try by using a SMILES with more than 1024 carbon atoms, and Open Babel seemed to generate the InChI without any complaints. Are we sure there is a 1024 atom limit? Maybe we can put in a check *after* generating the InChI if Open Babel doesn't have a problem generating one.

Reply:
This has changed a lot since we first wrote it, InChI has an implicit 1024 atom limit, but they may have relaxed it. As Open Babel is now running in a microservice it is less important to avoid triggering bad paths - maybe the Open Babel service dies and can we just restart it? I am fine with simply trying, and seeing if we get a string back. NextMove did some work on [larger molecule InChI](https://nextmovesoftware.com/blog/2014/11/26/introducing-new-formats-for-handling-macromolecules-smiles-and-inchi/) for example.

Reply:
I haven't seen the Open Babel microservice crash yet. Maybe I should try to crash it at some point to see what happens. But hopefully we can just restart it if it crashes.